packages/cli/src/controller/codegen-controller.ts

```diff
@@ -26,36 +39,102 @@ export interface processedField {
   name: string;
   type: string;
   required: boolean;
+  isArray: boolean;
+  isJsonInterface: boolean;
+}
+
+export async function generateJsonInterfaces(projectPath, schema: GraphQLSchema) {
+  const typesDir = path.join(projectPath, TYPE_ROOT_DIR);
+  const jsonObjects = getAllJsonObjects(schema);
+  const jsonInterfaces = jsonObjects.map((r) => {
+    const object = setJsonObjectType(r, jsonObjects);
+    const fields = processFields('jsonField', object.name, object.fields);
+    return {
+      interfaceName: object.name,
+      fields,
+    };
+  });
+
+  if (jsonInterfaces.length !== 0) {
+    const interfaceTemplate = {
+      props: {
+        jsonInterfaces,
+      },
+    };
+    try {
+      await renderTemplate(INTERFACE_TEMPLATE_PATH, path.join(typesDir, `interfaces.ts`), interfaceTemplate);
+      exportTypes.interfaces = true;
+    } catch (e) {
+      throw new Error(`When render json interfaces having problems.`);
+    }
+  }
 }
 
-export function processFields(className: string, fields: GraphQLEntityField[]): processedField[] {
+export function processFields(type: string, className: string, fields: GraphQLEntityField[] | GraphQLJsonFieldType[]) {
```

Comment:
can't find where `type` is used

Reply:
Two types 'entity' and 'jsonField' were pass though type for throw Error after transform field type , so the error message is more accurate. Also currently only 'entity' type need to extract indexed fields.